### PR TITLE
Added support for priority transaction manager. 

### PR DESCRIPTION
**Pull request by jason-bragg**
Only one priority transaction manager is allowed per transaction.  Priority TM be used to detect TOC in future changes.

Also moved ReadonlyCommit to resource interface from tm interface.  Call was placed in wrong interface when split.

### CODE REVIEW

File: src/Orleans.Transactions/DistributedTM/TransactionAgent.cs [hunk 1]
```diff
@@ -221,5 +213,25 @@ public void Abort(ITransactionInfo info, OrleansTransactionAbortedException reas
                  .Ignore();
             }
         }
+
+        // TODO: make overridable - jbragg
+        private KeyValuePair<ParticipantId,AccessCounter> SelectManager(TransactionInfo transactionInfo)
+        {
+            List<KeyValuePair<ParticipantId, AccessCounter>> priorityManagers = transactionInfo.Participants.SelectPriorityManagers().ToList();
```
Comment (xiazen) [inline src/Orleans.Transactions/DistributedTM/TransactionAgent.cs]: if there can only have 1 priority manager, then why `SelectPriorityManagers` returns multiple?

File: src/Orleans.Transactions/DistributedTM/TransactionInfo.cs [hunk 2]
```diff
@@ -39,8 +39,6 @@ public TransactionInfo(TransactionInfo other)
             this.IsReadOnly = other.IsReadOnly;
             this.TimeStamp = other.TimeStamp;
             this.Priority = other.Priority;
-            this.TMCandidate = other.TMCandidate;
-            this.TMBatchSize = other.TMBatchSize;
```
Comment (xiazen) [inline src/Orleans.Transactions/DistributedTM/TransactionInfo.cs]: I understand TMCandidate has been moved to TransactionInfo? But where are we setting TMBatchSize now? 
I guess my question is , why is this removed?

Comment (jason-bragg) [reply src/Orleans.Transactions/DistributedTM/TransactionInfo.cs]: Batch size was an optimization for TM selection.  It was not strictly necessary, but may influence performance.  I suspect it's a premature optimization, but we'll see.  If we get perf drop from this, we can revisit.

File: src/Orleans.Transactions/DistributedTM/ParticipantId.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        public enum Role
+        {
+            Resource = 1 << 0,
+            Manager = 1 << 1,
+            PriorityManager = 1 << 2
+        }
+
+        public string Name { get; }
+        public GrainReference Reference { get; }
+        public Role SupportedRoles { get; }
```
[the remark this answers is not in the document]
Comment (sebastianburckhardt) [reply src/Orleans.Transactions/DistributedTM/ParticipantId.cs]: It may be nice for application to be able to specify priority TM even if not using TOC. I certainly have encountered this need in my applications (e.g. for my TPC-C workload I wanted to control which grain is the TM).

File: src/Orleans.Transactions/DistributedTM/TransactionInfo.cs (hunk 2 above)
Comment (sebastianburckhardt) [reply src/Orleans.Transactions/DistributedTM/TransactionInfo.cs]: yep, we don't have super strong evidence that this optimization is important in general. I have seen it make a big difference in some cases but in the long run we may be better off just giving the application some control over TM selection than trying to do it automatically.

File: src/Orleans.Transactions/DistributedTM/ParticipantId.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebastianburckhardt) [reply src/Orleans.Transactions/DistributedTM/ParticipantId.cs]: I would keep it simple and just add a configuration option "PriorityTM" to the configuration object for the TransactionalState facet.
